ICSharpCode.Decompiler/CSharp/TranslatedExpression.cs

```diff
@@ -650,6 +650,10 @@ bool CastCanBeMadeImplicit(Resolver.CSharpConversions conversions, Conversion co
 				return newTargetType.IsKnownType(KnownTypeCode.FormattableString)
 					|| newTargetType.IsKnownType(KnownTypeCode.IFormattable);
 			}
+			if (conversion.IsUserDefined)
+			{
+				return conversions.IsImplicitSpanConversion(inputType, newTargetType);
```

Comment:
This change doesn't make any sense: you're not checking `oldTargetType` so this is essentially saying that any user-defined conversion, no matter which operator method is called, can be replaced with an implicit span conversion (if that's valid).
e.g. you could end up replacing a `int[]`->`MyClass` user-defined conversion with a `int[]`->`Span<int>` conversion.